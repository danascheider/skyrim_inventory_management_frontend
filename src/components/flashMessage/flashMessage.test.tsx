import { describe, test, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import FlashMessage from './flashMessage'

describe('FlashMessage', () => {
  describe('when there is a string message', () => {
    test('there is no list', () => {
      const wrapper = render(
        <FlashMessage type="info" hidden={false} message="Hello world." />
      )
      expect(wrapper).toBeTruthy()

      expect(screen.getByText('Hello world.')).toBeTruthy()
    })

    test('matches snapshot', () => {
      const wrapper = render(
        <FlashMessage type="info" hidden={false} message="Hello world." />
      )
      expect(wrapper).toMatchSnapshot()
    })

    describe('when there is a header', () => {
      test('the header and message are displayed', () => {
        render(
          <FlashMessage
            type="info"
            hidden={false}
            header="Greetings!"
            message="Hello world."
          />
        )

        expect(screen.getByText('Greetings!')).toBeTruthy()
        expect(screen.getByText('Hello world.')).toBeTruthy()
      })

      test('matches snapshot', () => {
        const wrapper = render(
          <FlashMessage
            type="info"
            hidden={false}
            header="Greetings!"
            message="Hello world."
          />
        )
        expect(wrapper).toMatchSnapshot()
      })
    })
  })

  describe('when there is an array message', () => {
    const msg = [
      'You will be assimilated.',
      'Your biological and technological distinctiveness will be added to our own.',
      'Resistance is futile.',
    ]

    test('displays all messages', () => {
      const wrapper = render(
        <FlashMessage type="info" hidden={false} message={msg} />
      )
      expect(wrapper).toBeTruthy()

      expect(screen.getByText('You will be assimilated.')).toBeTruthy()
      expect(
        screen.getByText(
          'Your biological and technological distinctiveness will be added to our own.'
        )
      ).toBeTruthy()
      expect(screen.getByText('Resistance is futile.')).toBeTruthy()
    })

    test('matches snapshot', () => {
      const wrapper = render(
        <FlashMessage type="info" hidden={false} message={msg} />
      )
      expect(wrapper).toMatchSnapshot()
    })

    describe('when there is a header', () => {
      test('displays the header text', () => {
        render(
          <FlashMessage
            type="info"
            hidden={false}
            header="We are the Borg."
            message={msg}
          />
        )

        expect(screen.getByText('We are the Borg.')).toBeTruthy()
        expect(screen.getByText('You will be assimilated.')).toBeTruthy()
        expect(
          screen.getByText(
            'Your biological and technological distinctiveness will be added to our own.'
          )
        ).toBeTruthy()
        expect(screen.getByText('Resistance is futile.')).toBeTruthy()
      })

      test('matches snapshot', () => {
        const wrapper = render(
          <FlashMessage
            type="info"
            hidden={false}
            header="We are the Borg."
            message={msg}
          />
        )
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
