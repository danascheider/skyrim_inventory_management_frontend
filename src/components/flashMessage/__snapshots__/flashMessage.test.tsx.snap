// Vitest Snapshot v1

exports[`FlashMessage > when there is a string message > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_91d025"
        style="--text-color: #4e6d83;"
      >
        Hello world.
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_91d025"
      style="--text-color: #4e6d83;"
    >
      Hello world.
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FlashMessage > when there is a string message > when there is a header > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_91d025"
        style="--text-color: #4e6d83;"
      >
        <p
          class="_header_91d025"
        >
          Greetings!
        </p>
        Hello world.
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_91d025"
      style="--text-color: #4e6d83;"
    >
      <p
        class="_header_91d025"
      >
        Greetings!
      </p>
      Hello world.
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FlashMessage > when there is an array message > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_91d025"
        style="--text-color: #4e6d83;"
      >
        <ul
          class="_list_91d025"
        >
          <li
            class="_msg_91d025"
          >
            You will be assimilated.
          </li>
          <li
            class="_msg_91d025"
          >
            Your biological and technological distinctiveness will be added to our own.
          </li>
          <li
            class="_msg_91d025"
          >
            Resistance is futile.
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_91d025"
      style="--text-color: #4e6d83;"
    >
      <ul
        class="_list_91d025"
      >
        <li
          class="_msg_91d025"
        >
          You will be assimilated.
        </li>
        <li
          class="_msg_91d025"
        >
          Your biological and technological distinctiveness will be added to our own.
        </li>
        <li
          class="_msg_91d025"
        >
          Resistance is futile.
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FlashMessage > when there is an array message > when there is a header > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_91d025"
        style="--text-color: #4e6d83;"
      >
        <p
          class="_header_91d025"
        >
          We are the Borg.
        </p>
        <ul
          class="_list_91d025"
        >
          <li
            class="_msg_91d025"
          >
            You will be assimilated.
          </li>
          <li
            class="_msg_91d025"
          >
            Your biological and technological distinctiveness will be added to our own.
          </li>
          <li
            class="_msg_91d025"
          >
            Resistance is futile.
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_91d025"
      style="--text-color: #4e6d83;"
    >
      <p
        class="_header_91d025"
      >
        We are the Borg.
      </p>
      <ul
        class="_list_91d025"
      >
        <li
          class="_msg_91d025"
        >
          You will be assimilated.
        </li>
        <li
          class="_msg_91d025"
        >
          Your biological and technological distinctiveness will be added to our own.
        </li>
        <li
          class="_msg_91d025"
        >
          Resistance is futile.
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
